drivers: wifi: siwx91x: Add features to opermode (boot config)

silabs-MuzaffarA opened the pull request:
- Added certain features to boot_config
- Added WIFI_SILABS_SIWX91X_LIMIT_PACKET_BUF_PER_STA to Kconfig
- Enabled roaming by default

jerome-pouiller commented on drivers/wifi/siwx91x/Kconfig.siwx91x:
```diff
@@ -45,8 +45,17 @@ config NET_MGMT_EVENT_STACK_SIZE
+	help
+		In AP mode, if the bit is set, only two packets per Station (STA)
+		would be buffered when the STA is in Power Save (PS) mode.
+		This helps manage buffer usage and ensures efficient packet handling.
```
The help message should be indented with one tab + 2 spaces.

jerome-pouiller commented on soc/silabs/silabs_siwx91x/siwg917/nwp.c:
```diff
@@ -35,8 +35,14 @@ int siwg91x_get_nwp_config(int wifi_oper_mode, sl_wifi_device_configuration_t *g
+				SL_SI91X_EXT_FEAT_DISABLE_XTAL_CORRECTION |
+				SL_SI91X_EXT_FEAT_NWP_QSPI_80MHZ_CLK_ENABLE |
+				SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0 |
+				SL_SI91X_EXT_FEAT_FRONT_END_INTERNAL_SWITCH,
```
You can align the values. Either:
```
			.custom_feature_bit_map = SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID |
						  SL_SI91X_CUSTOM_FEAT_RTC_FROM_HOST,
```
or:
```
			.custom_feature_bit_map = 
				SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID |
				SL_SI91X_CUSTOM_FEAT_RTC_FROM_HOST,
```

jhedberg replied:
One thing that might help some editors to auto-align this, is to put the value inside parentheses, i.e.:
```c
	.ext_custom_feature_bit_map = (SL_SI91X_EXT_FEAT_XTAL_CLK |
				       SL_SI91X_EXT_FEAT_1P8V_SUPPORT |
				       ...
				       SL_SI91X_EXT_FEAT_FRONT_END_INTERNAL_SWITCH),

jerome-pouiller reviewed (commented):
Only cosmetics.

Note that, theoretically, this the commit should have been split in three parts:
  - Enable roaming by default
  - Introduce `WIFI_SILABS_SIWX91X_LIMIT_PACKET_BUF_PER_STA`
  - Fix boot_config
 (don't bother with that for the current PR)

silabs-MuzaffarA committed:
drivers: wifi: siwx91x: Introduce flag for LIMIT_PACKET_BUF_PER_STA

Introduced WIFI_SILABS_SIWX91X_LIMIT_PACKET_BUF_PER_STA.
This flag limits packet queues in AP mode.

Signed-off-by: Muzaffar Ahmed <muzaffar.ahmed@silabs.com>

silabs-MuzaffarA committed:
drivers: wifi: siwx91x: Enable roaming by default

Set the default value for WIFI_SILABS_SIWX91X_ENABLE_ROAMING to y in
Kconfig.siwx91x

Signed-off-by: Muzaffar Ahmed <muzaffar.ahmed@silabs.com>

silabs-MuzaffarA committed:
drivers: wifi: siwx91x: Fix boot_config

Enable and rearrange some features in the boot config.

Signed-off-by: Muzaffar Ahmed <muzaffar.ahmed@silabs.com>

jerome-pouiller commented:
The commits title should be prefixed with `drivers: wifi: siwx91x: `

jerome-pouiller commented on soc/silabs/silabs_siwx91x/siwg917/nwp.c:
```diff
@@ -109,6 +109,11 @@ int siwg91x_get_nwp_config(int wifi_oper_mode, sl_wifi_device_configuration_t *g
 			LOG_WRN("Bluetooth is not supported in AP mode");
 		}
+
+		if (IS_ENABLED(WIFI_SILABS_SIWX91X_LIMIT_PACKET_BUF_PER_STA)) {
```
Does it works? I believe you mean `CONFIG_WIFI_SILABS_SIWX91X_LIMIT_PACKET_BUF_PER_STA`.

jerome-pouiller replied:
Not fixed.
